web-common/src/features/dashboards/dashboard-utils.ts

```diff
@@ -0,0 +1,70 @@
+import type {
+  QueryServiceMetricsViewComparisonToplistBody,
+  MetricsViewDimension,
+  MetricsViewMeasure,
+} from "@rilldata/web-common/runtime-client";
+import type { TimeControlState } from "./time-controls/time-control-store";
+import { getQuerySortType } from "./leaderboard/leaderboard-utils";
+import type { DashboardState_LeaderboardSortType } from "@rilldata/web-common/proto/gen/rill/ui/v1/dashboard_pb";
+
+export function isSummableMeasure(measure: MetricsViewMeasure): boolean {
+  return (
+    measure?.expression.toLowerCase()?.includes("count(") ||
+    measure?.expression?.toLowerCase()?.includes("sum(")
+  );
+}
+
+/**
+ * Returns a sanitized column name appropriate for use in e.g. filters.
+ *
+ * Even though this is a one-liner, we externalize it as a function
+ * becuase it is used in a few places and we want to make sure we
+ * are consistent in how we handle this.
+ */
+export function getDimensionColumn(dimension: MetricsViewDimension) {
+  return dimension?.column || dimension?.name;
+}
+
+export function prepareSortedQueryBody(
+  dimensionName: string,
+  measureNames: string[],
+  timeControls: TimeControlState,
+  sortMeasureName: string,
+  sortType: DashboardState_LeaderboardSortType,
+  sortAscending: boolean,
+  filterForDimension: {
```

Comment:
You can use `V1MetricsViewFilter` as a type here.

Reply:
ah thanks! fixed that